src/modules/wrapping/services/wrap.transactions.service.ts

```diff
@@ -15,36 +15,44 @@ export class WrapTransactionsService {
 
     async wrapEgld(sender: string, amount: string): Promise<TransactionModel> {
         const shardID = await this.mxProxy.getAddressShardID(sender);
-        const contract = await this.mxProxy.getWrapSmartContract(shardID);
-        return contract.methodsExplicit
-            .wrapEgld()
-            .withValue(amount)
-            .withGasLimit(gasConfig.wrapeGLD)
-            .withChainID(mxConfig.chainID)
-            .buildTransaction()
-            .toPlainObject();
+
+        return await this.mxProxy.getWrapSmartContractTransaction(
+            shardID,
+            new TransactionOptions({
+                function: 'wrapEgld',
+                chainID: mxConfig.chainID,
+                gasLimit: gasConfig.wrapeGLD,
+                sender: sender,
+                nativeTransferAmount: amount,
+            }),
+        );
     }
 
     async unwrapEgld(
         sender: string,
         amount: string,
     ): Promise<TransactionModel> {
-        const shardID = await this.mxProxy.getAddressShardID(sender);
-        const contract = await this.mxProxy.getWrapSmartContract(shardID);
-
-        const wrappedEgldToken = await this.wrapService.wrappedEgldToken();
+        const [shardID, wrappedEgldToken] = await Promise.all([
+            this.mxProxy.getAddressShardID(sender),
```

Comment:
by looking into this implementation:
```
async getAddressShardID(address: string): Promise<number> {
        const response = await this.getService().doGetGeneric(
            `address/${address}/shard`,
        );
        return response.shardID;
    }
```
there's no need for an API call for fetching the shard ID of an address. you can use `AddressComputer` from sdk-js-core:
```
const addressComputer = new AddressComputer();
let shard = addressComputer.getShardOfAddress(address);
```

Reply:
This is a good point, but I think it's outside the scope of this PR. I would create a separate PR for this, and update all the places where we use `getAddressShardID()`